docs/rules/jsx-sort-props.md

```diff
@@ -75,6 +76,24 @@ When `true`, alphabetical order is not enforced:
 <Hello tel={5555555} name="John" />
 ```
 
+### `reservedFirst`
+
+This can be a boolean or an array option.
+
+When `reservedFirst` is defined, React reserved props (`children`, `dangerouslySetInnerHTML`, `key`, `ref`) must be listed before all other props, but still respecting the alphabetical order:
```

Comment:
@ljharb tried working on this last night, ran into a small problem: what's the best way to identify if an AST node is a DOM element or not? I saw that `void-dom-elements-no-children` just creates a map of the ones it cares about, but for this we would care about all of them. Checking if the name starts with an upper-case letter would be a good guess, but would be wrong sometimes. It's not clear to me if `node.type` or if something in `node.attributes` can be used for this.

If I do just need to hardcode the list, that would mean we wouldn't catch custom DOM elements - is that ok?